### PR TITLE
Add a plot of Earth solid angle in radiator FOV

### PR DESCRIPTION
**Pull request by jzuhone**
This PR adds a plot of the Earth solid angle in the ACIS radiator FOV to the focal plane temperature pages. The curve is added to the off-nominal roll plot.

An example here:

http://cxc.cfa.harvard.edu/acis/tmp/earth_solid_angle

This required:

- Overriding the `_make_state_plots` method to add the data to the roll plot
- Overriding `write_temps` so that the Earth solid angle data could be written to the `temperatures.dat` file. 
- making `perigee_passages` an attribute of the `ACISFPCheck` object so that it can be used to make the perigee passages on all of the prediction plots

For this to work, PR acisops/acis_thermal_check#19 is also required.

### CODE REVIEW

**Review by richardjedgar — APPROVED** [review]
Based on looking at sample outputs, I like this.

File: acisfp_check/acisfp_check.py [hunk 1]
```diff
@@ -492,6 +539,35 @@ def search_obsids_for_viols(self, limit_name, limit, observations, temp, times,
...
+        e = self.predict_model.comp['earthheat__fptemp'].dvals
+        temp_recs = [(times[i], DateTime(times[i]).date, T[i], e[i])
+                     for i in range(len(times))]
+        temp_array = np.rec.fromrecords(
+            temp_recs, names=('time', 'date', self.msid, 'earth_fov'))
+        fmt = {self.msid: '%.2f',
+               'time': '%.2f',
+               'earth_solid_angle': '%.7f'}
+        out = open(outfile, 'w')
+        Ska.Numpy.pprint(temp_array, fmt, out)
```
Comment (jeanconn) [inline acisfp_check/acisfp_check.py]: I don't have a problem with this, but I've been moving to using astropy.table for all of our table printing needs.

Comment (jzuhone) [reply acisfp_check/acisfp_check.py]: This is just a slight modification to what the default acis_thermal_check code is already doing, which is why it uses the old table writing. We can probably switch to AstroPy tables at some point in the near future.

Comment (Gregg140) [reply acisfp_check/acisfp_check.py]: This is a deviation from the standard format of temperatures.dat files that has been in existence for years. And it's a deviation that's specific to focal plane temperature model temperatures.dat files. So lots of people are not expecting this.

I don't have any problem with that per se.  But we have to be sure everyone who might use the files in other code are aware of the change.  We want to minimize the chances someone's code that, heretofore has worked, suddenly stops working. Given that this is a new column placed on the end, the chances are minimized that something will break but not zero.

Comment (jzuhone) [reply acisfp_check/acisfp_check.py]: The PSMC temperatures.dat file used to have pin1at in there, so this isn’t a completely new thing. 

I am pretty sure it is only us and SOT MP that care about this file so I will make sure to alert them to this.

File: acisfp_check/acisfp_check.py [hunk 2]
```diff
@@ -492,6 +539,35 @@ def search_obsids_for_viols(self, limit_name, limit, observations, temp, times,
...
+        times : NumPy array
+            Times in seconds from the start of the mission
+        temps : NumPy array
+            Temperatures in Celsius
+        """
+        outfile = os.path.join(outdir, 'temperatures.dat')
+        mylog.info('Writing temperatures to %s' % outfile)
+        T = temps[self.name]
+        e = self.predict_model.comp['earthheat__fptemp'].dvals
+        temp_recs = [(times[i], DateTime(times[i]).date, T[i], e[i])
```
Comment (taldcroft) [inline acisfp_check/acisfp_check.py]: It's not a hard rule, but I've pretty much moved away from recarray to either np structured array or astropy Table.

I think there are actual performance / memory / readability benefits to using Table here (not that performance really matters in practice here).  (See http://cxc.harvard.edu/mta/ASPECT/tool_doc/pydocs/Chandra.Time.html#fast-conversion-functions for `secs2date`).
```
temp_table = Table([times, secs2date(times), T, e], names=[...], copy=False)
temp_table[self.msid].format = '.2f'
..
temp_table.write(out, format='ascii', overwrite=True)
```

Comment (jzuhone) [reply acisfp_check/acisfp_check.py]: I’m happy to change it—I will note that this is almost an identical copy of write_temps in acis_thermal_check aside from the extra column, so this is not new code.

File: acisfp_check/acisfp_check.py (hunk 1 above)
Comment (taldcroft) [reply acisfp_check/acisfp_check.py]: Plan B might be a new file `earth_solid_angles.dat`.  Then SOT MP would not need to change their code (assuming that is the case now?).

Comment (Gregg140) [reply acisfp_check/acisfp_check.py]: We don't know that SOT MP has code that would need to be changed. I'm just raising the issue so it gets thought over.

File: acisfp_check/acisfp_check.py (hunk 2 above)
Comment (jzuhone) [reply acisfp_check/acisfp_check.py]: @taldcroft I should note that the version of AstroPy on flight ska2 is too old for this code--`Table.write` doesn't recognize the `overwrite` keyword and there are some issues with the format strings.

**Pull request merged**